plugins/dac_data_manager.c

```diff
@@ -1237,23 +1227,46 @@ static void gui_manager_create(struct dac_data_manager *manager)
 	gtk_widget_show(hbox);
 }
 
+/* Uses the "TX*_I|Q_F* name convention to build a name for the channel at the given index.
+ * E.g. TX1_I_F1 - 0 (the index)
+ *      TX1_I_F2 - 1
+ *      TX1_Q_F1 - 2
+ *      TX1_Q_F2 - 3
+ *      TX2_I_F1 - 4
+ *      TX2_I_F2 - 5
+ *      ...
+ * Returns a newlly alocated char array.
+ */
+static char *build_default_channel_name_from_index(guint ch_index)
+{
+	guint tx_index = (ch_index / TX_NB_TONES) + 1; /* TX couting starts from 1 (not 0) */
+	guint tone_index = (ch_index % 2) + 1; /* There are always 2 tones (I/Q). Count starts at 1 */
+	const char i_q_type = (ch_index & 0x02) ? Q_CHANNEL : I_CHANNEL; /* First two indexes are I, next two are Q and so on*/
+
+	return g_strdup_printf("TX%i_%c_F%i", tx_index, i_q_type, tone_index);
+}
+
 #define TONE_ID "altvoltage"
 #define TONE_ID_SIZE (sizeof(TONE_ID) - 1)
 
-static const char * get_tone_name(struct iio_channel *ch)
+/* Returns a newlly allocated tone name for the given channel in case of
+ * success and NULL otherwise.
+ */
+static char * get_tone_name(struct iio_channel *ch)
 {
-	const char *name;
+	char *name;
 	char tone_index;
 
-	name = iio_channel_get_name(ch);
+	name = g_strdup( iio_channel_get_name(ch));
 
 	/* If name convention "TX*_I|Q_F* is missing */
 	if (name && strncmp(name, "TX", 2) != 0) {
-		name = iio_channel_get_id(ch);
+		g_free(name);
+		name = g_strdup(iio_channel_get_id(ch));
 		if (name && !strncmp(name, TONE_ID, TONE_ID_SIZE)) {
 			tone_index = name[TONE_ID_SIZE];
 			if (tone_index && g_ascii_isdigit(tone_index))
-				name = default_channel_names[tone_index - '0'];
+				name = build_default_channel_name_from_index(tone_index - '0');
```

Comment:
another approach, would be to use a `name_buf[32]` which we could pass to `build_default_channel_name_from_index(name_buf, tone_index - '0'`)

that way, there shouldn't be a need to do strdup()

Reply:
If I do that, then I need to carry a reference of the dac_data_manager all the way to this function. I don't think that it will be any better than the way it is now.

Reply:
another approach would be to build the list once on the module constructor and free it on the module destructor path;
or do, some sort of caching inside `build_default_channel_name_from_index`; 

[atm] i am not questioning whether all `strdup()`s are all free'd here; i'm questioning whether someone else who will want to extend/change this code [later], will remember to free them [given this `strdrup()` + `free()` pattern]